build/lib/src/builder/builder.dart

```diff
@@ -34,18 +34,24 @@ class BuilderOptions {
   /// `bool` or `List` or `Map` of these types.
   final Map<String, dynamic> config;
 
-  const BuilderOptions(this.config);
+  /// Whether or not this builder is running on the root package.
+  final bool isRoot;
+
+  const BuilderOptions(this.config, {bool isRoot}) : isRoot = isRoot ?? false;
 
   /// Returns a new set of options with keys from [other] overriding options in
   /// this instance.
   ///
-  /// Values are overridden at a per-key granularity. There is no value level
-  /// merging. [other] may be null or empty, in which case this instance is
-  /// returned directly.
-  BuilderOptions overrideWith(BuilderOptions other) =>
-      other == null || other.config.isEmpty
-          ? this
-          : new BuilderOptions({}..addAll(config)..addAll(other.config));
+  /// Config values are overridden at a per-key granularity. There is no value
+  /// level merging. [other] may be null or empty, in which case this instance
```

Comment:
no longer returned directly if other is empty. (Also the meaning of "empty" might not make sense anymore)

do we want to do any other checking for the case where `other.config` is empty and `other.isRoot` is already the same as `this.isRoot`?

Reply:
I don't think it is worth optimizing really, these objects are cheap